libraries/NetworkClientSecure/src/ssl_client.h

```diff
@@ -26,6 +26,10 @@ typedef struct sslclient_context {
 
   unsigned long socket_timeout;
   unsigned long handshake_timeout;
+
+  int last_error = 0;
+  int peek_buf = -1;
```

Comment:
you are setting here default value `-1` and the code calls `memset(ssl_client, 0, sizeof(sslclient_context));` which would set it to 0. That is why you are getting the error. Do not add default value like that and  make sure that when cleared it is then set properly.